kyaml/setters2/settersutil/fieldsetter_test.go

```diff
@@ -26,81 +27,94 @@ func TestSetAllSetterDefinitions(t *testing.T) {
           name: replicas
           value: "4"`
 
-	destFile1 := `apiVersion: apps/v1
+	destFile := `apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: nginx-deployment
   namespace: some-other-namespace # {"$ref": "#/definitions/io.k8s.cli.setters.namespace"}
 spec:
-  replicas: 3 # {"$ref": "#/definitions/io.k8s.cli.setters.replicas"}`
+  replicas: 3 # {"$ref": "#/definitions/io.k8s.cli.setters.replicas-new"}`
 
-	destFile2 := `apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: nginx-deployment
-  namespace: some-other-namespace2 # {"$ref": "#/definitions/io.k8s.cli.setters.namespace"}
-spec:
-  replicas: 2 # {"$ref": "#/definitions/io.k8s.cli.setters.replicas"}`
+	destOpenAPI := `openAPI:
+  definitions:
+    io.k8s.cli.setters.namespace:
+      x-k8s-cli:
+        setter:
+          name: namespace
+          value: "some-other-namespace"
+    io.k8s.cli.setters.replicas-new:
+      x-k8s-cli:
+        setter:
+          name: replicas-new
+          value: "3"`
 
 	expectedDestFile := `apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: nginx-deployment
   namespace: project-namespace # {"$ref": "#/definitions/io.k8s.cli.setters.namespace"}
 spec:
-  replicas: 4 # {"$ref": "#/definitions/io.k8s.cli.setters.replicas"}`
+  replicas: 3 # {"$ref": "#/definitions/io.k8s.cli.setters.replicas-new"}`
 
-	srcDir, err := ioutil.TempDir("", "")
-	if !assert.NoError(t, err) {
-		t.FailNow()
-	}
+	expectedDestOpenAPIFile := `openAPI:
+  definitions:
+    io.k8s.cli.setters.namespace:
+      x-k8s-cli:
+        setter:
+          name: namespace
+          value: "project-namespace"
+          isSet: true
+    io.k8s.cli.setters.replicas-new:
+      x-k8s-cli:
+        setter:
+          name: replicas-new
+          value: "3"`
 
-	destDir1, err := ioutil.TempDir("", "")
+	srcDir, err := ioutil.TempDir("", "")
 	if !assert.NoError(t, err) {
 		t.FailNow()
 	}
 
-	destDir2, err := ioutil.TempDir("", "")
+	destDir, err := ioutil.TempDir("", "")
 	if !assert.NoError(t, err) {
 		t.FailNow()
 	}
 
 	defer os.RemoveAll(srcDir)
-	defer os.RemoveAll(destDir1)
-	defer os.RemoveAll(destDir2)
+	defer os.RemoveAll(destDir)
 
-	err = ioutil.WriteFile(srcDir+"/OpenAPIFile", []byte(srcOpenAPIFile), 0600)
+	err = ioutil.WriteFile(filepath.Join(srcDir, "Krmfile"), []byte(srcOpenAPIFile), 0600)
 	if !assert.NoError(t, err) {
 		t.FailNow()
 	}
-	err = ioutil.WriteFile(destDir1+"/destFile.yaml", []byte(destFile1), 0600)
+	err = ioutil.WriteFile(filepath.Join(destDir, "destFile.yaml"), []byte(destFile), 0600)
 	if !assert.NoError(t, err) {
 		t.FailNow()
 	}
-
-	err = ioutil.WriteFile(destDir2+"/destFile.yaml", []byte(destFile2), 0600)
+	err = ioutil.WriteFile(filepath.Join(destDir, "Krmfile"), []byte(destOpenAPI), 0600)
 	if !assert.NoError(t, err) {
 		t.FailNow()
 	}
 
-	err = SetAllSetterDefinitions(srcDir+"/OpenAPIFile", destDir1, destDir2)
+	err = SetAllSetterDefinitions(true, filepath.Join(srcDir, "Krmfile"), destDir)
```

Comment:
Seems like we only test this for one example with the `syncOpenAPI` parameter set to true. Seems like there must be quite a few interesting cases for this we should be testing.

Reply:
Added another case where it is false.